src/lua/digest.lua

```diff
@@ -267,4 +313,70 @@ m['aes256cbc'] = {
     end
 }
 
+for _, var in ipairs({'32', '64'}) do
+    local xxHash
+
+    local update_fn_name = 'XXH' .. var .. '_update'
+    local digest_fn_name = 'XXH' .. var .. '_digest'
+    local reset_fn_name = 'XXH' .. var .. '_reset'
+    local copy_fn_name = 'XXH' .. var .. '_copyState'
+    local methods = {
+        default_seed = 0,
+
+        update = function(self, str)
+            if type(str) ~= 'string' then
+                error("Usage xxhash" .. var .. ":update(string)")
```

Comment:
I would propose to raise the error in the caller's context, via the second argument of `error()` function. To raise it with the caller's file and line, which should be more useful. Up to you.

Reply:
Changed to error(..., 2).

```
./src/tarantool file.lua 
LuajitError: file.lua:2: Usage digest.xxhash32(string[, unsigned number])
```